File: FrontEnd/foyer/src/app/services/announcement.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Announcement, AnnouncementType } from '../models/announcement.model';


@Injectable({
  providedIn: 'root'
})
export class AnnouncementService {
  private apiUrl = 'http://localhost:8086/api/announcements';

  constructor(private http: HttpClient) { }

  getAllAnnouncements(): Observable<Announcement[]> {
    return this.http.get<Announcement[]>(this.apiUrl);
  }

  getAnnouncementById(id: number): Observable<Announcement> {
    return this.http.get<Announcement>(`${this.apiUrl}/${id}`);
  }

  getAnnouncementsByType(type: AnnouncementType): Observable<Announcement[]> {
    return this.http.get<Announcement[]>(`${this.apiUrl}/type/${type}`);
  }

  getActiveAnnouncements(): Observable<Announcement[]> {
    return this.http.get<Announcement[]>(`${this.apiUrl}/active`);
  }

  createAnnouncement(announcement: Announcement): Observable<Announcement> {
    return this.http.post<Announcement>(this.apiUrl, announcement);
  }

  updateAnnouncement(id: number, announcement: Announcement): Observable<Announcement> {
    return this.http.put<Announcement>(`${this.apiUrl}/${id}`, announcement);
  }

  deleteAnnouncement(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

  deactivateAnnouncement(id: number): Observable<void> {
    return this.http.put<void>(`${this.apiUrl}/${id}/deactivate`, {});
  }
} 

File: BackEnd/Microservices/announcement/src/main/java/tn/esprit/announcement/service/AnnouncementService.java
package tn.esprit.announcement.service;

import tn.esprit.announcement.entity.Announcement;
import tn.esprit.announcement.entity.AnnouncementType;

import java.util.List;
import java.util.Optional;

public interface AnnouncementService {
    Announcement createAnnouncement(Announcement announcement);
    Optional<Announcement> getAnnouncementById(Long id);
    List<Announcement> getAllAnnouncements();
    List<Announcement> getAnnouncementsByType(AnnouncementType type);
    List<Announcement> getActiveAnnouncements();
    Announcement updateAnnouncement(Long id, Announcement announcement);
    void deleteAnnouncement(Long id);
    void deactivateAnnouncement(Long id);
}
